tests/traced_data/test_traced_data.py

```diff
@@ -58,6 +58,33 @@ def test_append_data(self):
         history = td.get_history("gender")
         self.assertListEqual(list(map(lambda x: x["value"], history)), ["man", "male"])
 
+    def test_hide_keys(self):
+        td = self.generate_test_data()
+
+        self.assertEqual(td["id"], "0")
+        self.assertEqual(td["phone"], "+441632000001")
+        self.assertEqual(td["gender"], "man")
+
+        td.hide_keys({"gender", "phone"}, Metadata("test_user", "hide_keys", time.time()))
+
+        self.assertTrue("id" in td)
+        self.assertFalse("phone" in td)
+        self.assertFalse("gender" in td)
+
+        self.assertRaises(KeyError, lambda: td["gender"])
+        self.assertRaises(KeyError, lambda: td["phone"])
+
+        self.assertEqual(td.get("gender"), None)
+        self.assertEqual(td.get("id"), "0")
+
+        self.assertSetEqual(set(td.keys()), {"id"})
+        self.assertDictEqual(dict(td.items()), {"id": "0"})
+        self.assertSetEqual(set(td.values()), {"0"})
+        self.assertEqual(len(td), 1)
+
+        with self.assertRaisesRegex(KeyError, "gender"):
```

Comment:
Could you add a test to ensure that:

-> Generate data with key (gender)
assert(Has value)
-> Hide key (gender)
assert(Has doesn't have value)
-> Add value (gender)
assert(Has value)

I.e. make sure that the we can re add a key over a hidden key and that it works

and another test where hiding a key that doesn't exist raises an exception

Reply:
added